syntax = "proto3";

enum COMM_CMD {
    QSPI_WRITE = 0;
    QSPI_ERASE = 2;
    QSPI_MASS_ERASE = 5;
    BOOTLOADER_INTERCEPT = 10;
}

enum COMM_RES {
    OK = 0;
    ERR_UNKNOWN_CMD = 1;
    ERR_PARSE_RQ = 2;
    ERR_QSPI_ABORT = 3;
    ERR_QSPI_WRITE = 4;
    ERR_QSPI_ERASE = 6;
}

message CommCmdBasicRp {
    uint32 result = 1;
}

message CommCmdQspiWriteRq {
    uint32 addr = 1;
    bytes buff = 2;
}

message CommCmdQspiEraseRq {
    uint32 addr = 1;
    uint32 size = 2;
}

message CommCmdQspiMassEraseRq {
}

message CommCmdBootloaderInterceptRq {
    uint32 intercept = 1; // 0 = off, 1 = on
}
